core/arch/arm/mm/tee_mm.c

```diff
@@ -190,13 +198,13 @@ tee_mm_entry_t *tee_mm_alloc(tee_mm_pool_t *pool, size_t size)
 
 			if (remaining < size)
 				/* out of memory */
-				return NULL;
+				goto out;
 		}
 	}
 
-	nn = tee_mm_add(entry);
+	nn = tee_mm_add(entry, next);
```

Comment:
`tee_mm_add()` now returns the value of `next`. Maybe it should simply return `void` and code below (after to now useless test of `nn`) should use `next` instead of `nn`. `actually, `nn` could be fully removed.